compare.py

```diff
@@ -0,0 +1,413 @@
+# import the necessary packages
+import codecs
+import cv2
+import imutils
+import itertools
+import json
+import logging
+import numpy as np
+import os
+import pickle
+import pytesseract
+import skimage
+from datetime import date
+from datetime import datetime
+from datetime import timedelta
+from jira import JIRA, JIRAError
+from skimage.measure import compare_ssim
+from skimage.feature import match_template
+from sys import platform
+debug = False
+
+
+def compare(folder, datum_image="imageB.png", latest_image="imageA.png",
+            masked_datum="imageB_masked.png"):
+
+    datum_image, gray_datum_image, latest_image, gray_latest_image = preprocess_images(folder=folder,
+                                                                                       datum_image=datum_image,
+                                                                                       latest_image=latest_image)
+    contours = extract_contours(gray_datum_image, gray_latest_image)
+
+    rectangles = extract_regions_of_interest(contours=contours, gray_datum_image=gray_datum_image,
+                                             gray_latest_image=gray_latest_image)
+
+    jira_ticket = None
+
+    if len(rectangles) > 0:
+        merged_rectangles = combine_boxes(boxes=rectangles)
+        new_text = extract_text(boxes=rectangles, image=gray_latest_image)
+        diff_text = save_text(folder=folder, text=new_text)
+
+        before_and_after_png_path = save_images(folder=folder, datum_image=datum_image, latest_image=latest_image)
+
+        jira_ticket = {'summary': "Website change detected at {}".format(folder.split(os.sep)[-1]),
+                       'description': "Changes detected at: {}\nPlease see the attachments for details.".format(folder),
+                       'attachments': [diff_text, before_and_after_png_path, "{}{}{}".format(folder, os.sep, masked_datum)]}
+
+    logging.info("\tChange detected: {}".format(len(rectangles) > 0))
+
+    return jira_ticket
+
+
+def save_images(folder, datum_image, latest_image):
+
+    # save the output image
+    diff_png = "{}{}{}".format(folder, os.sep, "diff.png")
+    logging.debug("Diff image saved to {}".format(diff_png))
+
+    cv2.imwrite(diff_png, latest_image)
+    before_and_after_png = "{}{}{}".format(folder, os.sep, "before_and_after.png")
+
+    logging.info("\tBefore and after png saved to: {}".format(before_and_after_png))
+    cv2.imwrite(before_and_after_png, tile_images(left_image=datum_image, right_image=latest_image))
+
+    return before_and_after_png
+
+
+def save_text(folder, text, filename="diff.txt"):
+
+    path = "{}\{}".format(folder, filename)
+    f = codecs.open(path, 'w', encoding='utf8')
+
+    for text in text:
+        logging.debug("Extracted the following text from: [{}]".format(text))
```

Comment:
we should be able to maintain logging format for levels(i.e error, debug, info etc) , need logging formatter method for log handler .